whylog/log_reader/read_utils.py

```diff
@@ -0,0 +1,61 @@
+from whylog.log_reader.exceptions import EmptyFile, OffsetBiggerThanFileSize
+
+
+class ReadUtils(object):
+    @classmethod
+    def _read_content(cls, fd, position, buf_size):
+        fd.seek(position)
+        return fd.read(buf_size)
+
+    @classmethod
+    def _read_split_lines(cls, fd, position, buf_size):
+        content = cls._read_content(fd, position, buf_size)
+        return content.split('\n')
+
+    @classmethod
+    def _join_results(cls, first_part, second_part):
+        if not first_part:
+            if not second_part:
+                return []
+            return second_part
+        if not second_part:
+            return first_part
+        return first_part[:-1] + ["".join((first_part[-1], second_part[0]))] + second_part[1:]
+
+    @classmethod
+    def _expand_after(cls, fd, position):
+        fd.seek(position)
+        line = fd.readline()
+        if not line:
+            raise OffsetBiggerThanFileSize(position)
+        return line.strip()
```

Comment:
whitespace can be meaningful for someone, you can't just drop part of the data like that. If there is a trailing newline or something, use offsets to deal with it